solrman/smstorage/zk_storage.go

```diff
@@ -205,51 +206,81 @@ func (s *ZkStorage) GetStationaryOrgList() ([]string, error) {
 	return children, nil
 }
 
-func (s *ZkStorage) IsDisabled() (bool, string) {
+func (s *ZkStorage) IsDisabled() (bool, error) {
 	path := s.disabledPath()
-	data, _, err := s.conn.Get(path)
+	children, _, err := s.conn.Children(path)
 	if err != nil {
-		// If no node, it is not disabled.
-		if err == zk.ErrNoNode {
-			return false, ""
-		}
+		s.logger.Errorf("could not check the children at %s in ZK: %s", path, err)
+		return true, err
```

Comment:
it should return false when there is err ?

Reply:
No, I specifically want it to return "disabled" if it is erroring. If it's meant to be disabled, it's safer to return disabled in the event of an error. That will alert. A false-enabled would be much more trouble.